tests/unit/test_cli.py

```diff
@@ -347,6 +347,20 @@ def test_env_filter_key(cli_runner, vault_with_token, mocker):
     assert kwargs["environ"]["PASSWORD"] == "xxx"
 
 
+def test_env_omit_single_key(cli_runner, vault_with_token, mocker):
+    exec_command = mocker.patch("vault_cli.environment.exec_command")
+
+    vault_with_token.db = {"foo/bar": {"value": "yay"}, "foo/baz": {"password": "yo"}}
+    cli_runner.invoke(
+        cli.cli, ["env", "--path", "foo", "--omit-single-key", "--", "echo", "yay"]
+    )
+
+    _, kwargs = exec_command.call_args
+    assert kwargs["command"] == ("echo", "yay")
+    assert kwargs["environ"]["FOO_BAR"] == "yay"
+    assert kwargs["environ"]["FOO_BAZ"] == "yo"
```

Comment:
Not sure if it would be preferable to just check that `get_envvars_for_secrets` is called with the right parameter, because given the file name, it's a unit test, not an integration one. But in reality, even if it's in `units/` this is an integration test in disguise so let it be.
:+1:

Reply:
this is not the only "integration test in disguise here" unfortunately, let's keep it that way